File: Leetcode/python/sol1984.py
# https://leetcode.com/problems/minimum-difference-between-highest-and-lowest-of-k-scores/description/
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        if k <= 1:
            return 0
        nums.sort()
        result = float('inf')
        for i in range(len(nums ) - k +1):
            result = min(result, nums[ i + k -1] - nums[i])
        return result

File: Leetcode/typescript/sol1984.ts

// https://leetcode.com/problems/minimum-difference-between-highest-and-lowest-of-k-scores/description/?envType=daily-question&envId=2026-01-25

function minimumDifference(nums: number[], k: number): number {
    nums.sort((a, b) => a - b);
    let result = Infinity;
    for(let i = 0;i<nums.length-k+1; i++){
        result = Math.min(result, nums[i+k-1] - nums[i])
    }
    return result;
};